src/Enum/DocumentType.php

```diff
@@ -0,0 +1,48 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Valantic\ElasticaBridgeBundle\Enum;
+
+use Pimcore\Model;
+
+enum DocumentType: string
+{
+    case ASSET = 'asset';
+
+    case DOCUMENT = 'document';
+
+    case DATA_OBJECT = 'object';
+
+    case VARIANT = 'variant';
+
+    /**
+     * @return class-string
+     */
+    public function baseClass(): string
+    {
+        return match ($this) {
+            self::ASSET => Model\Asset::class,
+            self::DOCUMENT => Model\Document::class,
+            self::DATA_OBJECT, self::VARIANT => Model\DataObject::class,
+        };
+    }
+
+    /** @return self[] */
+    public static function casesDataObjects()
```

Comment:
You could also add `array` as function return type. But ok, it's declared in the `@retunr` comment